[NUI] Add Window documents (#1170)
Samsung/tizen-docs: docs/application/dotnet/guides/nui/window.md

{
    if (e.Touch.GetState(0) == PointStateType.Down)
    {
        // The window has been touched, do something
    }
}
```

### Key event

Key event is emitted when the window receives a key event from the window manager.

The following code shows how to handle the key event:

```csharp
window.KeyEvent += OnWindowKeyEvent;

private void OnWindowKeyEvent(object sender, Window.KeyEventArgs e)
{
    if (e.Key.State == Key.StateType.Down)
    {
        if (e.Key.KeyPressedName == "Left")
        {
            // The left arrow key is pressed, do something
        }
        else if (e.Key.KeyPressedName == "Right")
        {
            // The right arrow key is pressed, do something
        }
    }
}
```

### Resize event

Resize event is emitted when the window is resized.

The following code shows how to handle the resize event:

```csharp
window.Resized += OnWindowResized;

private void OnWindowResized(object sender, Window.ResizedEventArgs e)
{
    // Window is resized, do something
}
```

For more information, see [Window](https://samsung.github.io/TizenFX/latest/api/Tizen.NUI.BaseComponents.View.html) API.

## Window rotation
To support rotation, NUI Provides several API.

> The following are how to use them:

Reviewer: @mijin85cho, I agree with your suggestion.

You can specify the initial size and position of the default window using the constructor of the application:
```csharp
public MyApplication(Size2D windowSize, Position2D windowPosition) : base(windowSize, windowPosition)
{
}

MyApplication myApp = new MyApplication(new Size2D(1920, 1080), new Position2D(0, 0));
```

The instance of the default window can be retrieved:

```csharp
Window window = Window.Instance;
```

The window itself is just a blank container, and the views can be added to or removed from the window:

```csharp
// Create a new view
View view = new View();

// Add the view to the window
window.Add(view);

// Remove the view from the window
window.Remove(view);
```

The following code shows how to add content to the window:

```csharp
// Get the window instance
Window window = Window.Instance;
window.BackgroundColor = Color.White;

// Create a TextLabel
TextLabel title = new TextLabel("Hello World");
title.Size2D = new Size2D(window.Size.Width, window.Size.Height);
title.HorizontalAlignment = HorizontalAlignment.Center;
title.VerticalAlignment = VerticalAlignment.Center;

// Add the text to the window
window.Add(title);
```

![ NewWindow ](./media/new_window.png)

> 

Reviewer 2: Question. For now "Deleting Window" section is removed. 

So now in this page, we have description of how to create a window, but don't have description of how to remove a window.

I think updating "Delete Window" section is better than removing it. Would you please let me your opinion?

Author: API for deleting window is not public, so i think document for deleting window need to be added other ACR

## Resize window

> The size of a window can be retrieved, and the window can be resized.

Reviewer: ```suggestion
The size of a window can be retrieved, and the window can be resized:
```

```csharp
// Retrieve the current size of the window
Size2D windowSize = window.WindowSize;

// Resize the window by increasing its width
windowSize.Width += 100;
window.WindowSize = windowSize;
```

When the window size is changed, a resize event is emitted.

## Move window

> It is easy to retrieve the current position of a window and move the window to a new position.

Reviewer: ```suggestion
It is easy to retrieve the current position of a window and move the window to a new position:
```

```csharp
// Retrieve the current position of the window
Position2D windowPosition = window.WindowPosition;

// Moving the window to the right
windowPosition.X += 100;
window.WindowPosition = windowPosition;
```

## Show and hide window

> When a window is created, by default it is in visible state. It is also possible to hide a window by making it invisible.

Reviewer: ```suggestion
When a window is created, by default it is in visible state. It is also possible to hide a window by making it invisible:
```

    {
        // The window has been touched, do something
    }
}
```

### Key event

Key event is emitted when the window receives a key event from the window manager.

The following code shows how to handle the key event:

```csharp
window.KeyEvent += OnWindowKeyEvent;

private void OnWindowKeyEvent(object sender, Window.KeyEventArgs e)
{
    if (e.Key.State == Key.StateType.Down)
    {
        if (e.Key.KeyPressedName == "Left")
        {
            // The left arrow key is pressed, do something
        }
        else if (e.Key.KeyPressedName == "Right")
        {
            // The right arrow key is pressed, do something
        }
    }
}
```

### Resize event

Resize event is emitted when the window is resized.

The following code shows how to handle the resize event:

```csharp
window.Resized += OnWindowResized;

private void OnWindowResized(object sender, Window.ResizedEventArgs e)
{
    // Window is resized, do something
}
```

For more information, see [Window](https://samsung.github.io/TizenFX/latest/api/Tizen.NUI.BaseComponents.View.html) API.

## Window rotation

To support rotation, NUI provides two methods.
// To do: tscholb

> ### Enable or disable window

Reviewer 2: @surbhig12 I want to ask your opinion about this heading.
I reverted back to "window", but not sure whether it is proper word for this section.
According to the content, still it seems it is not containing the exact meaning.

In my opinion, another suggestion can be below. Would you please let me know your opinion also?

\### Enable or disable orientations

If the device supports rotation, application can rotate the window with specific orientation.

The supported orientation are `Portrait`, `Landscape`, `PortraitInverse` and `LandscapeInverse`.
`Portrait` means the height is larger than the width or same.
`Landscape` means the width is larger than the height.

<img src="./media/device_rotate2.png">

> Application can enable or disable a specific orientation to use orientation functions. The following code shows that `Portrait`, `Landscape`, `PortraitInverse` and `LandscapeInverse` are supported in the application.
> If the device's width is larger than the height, then the window will be shown as landscape when application is launched. If the device's width is smaller than the height or same, then the window will be shown as portrait.

Reviewer: Please add colon instead of a period at the end of the sentence.

```suggestion
Application can enable or disable a specific orientation to use orientation functions. The following code shows that `Portrait`, `Landscape`, `PortraitInverse` and `LandscapeInverse` are supported in the application.
If the device's width is larger than the height, then the window will be shown as landscape when application is launched. If the device's width is smaller than the height or same, then the window will be shown as portrait:
```

```csharp
// To enable Portrait
window.AddAvailableOrientation( Portrait );

// To enable Landscape
window.AddAvailableOrientation( Landscape );

// To disable PortraitInverse
window.RemoveAvailableOrientation( PortraitInverse );

// To enable LandscapeInverse
window.AddAvailableOrientation( LandscapeInverse );
```

### Preferred orientation

> If you want the window to appear in a specific orientation, the preferred orientation function is supported. The preferred orientation is one of the available orientations. For example, as shown in the following code, the available orientation list has `Portrait`, `Landscape`, and `LandscapeInverse`, and the preferred orientation is set to Landscape. This implies that the window will appear in landscape even when the device is rotated with any other orientation.

Reviewer: Please add colon instead of a period at the end of the sentence.

```suggestion
If you want the window to appear in a specific orientation, the preferred orientation function is supported. The preferred orientation is one of the available orientations. For example, as shown in the following code, the available orientation list has `Portrait`, `Landscape`, and `LandscapeInverse`, and the preferred orientation is set to Landscape. This implies that the window will appear in landscape even when the device is rotated with any other orientation:
```

```csharp
// To enable available orientations
window.AddAvailableOrientation( Portrait );
window.AddAvailableOrientation( Landscape );
window.AddAvailableOrientation( LandscapeInverse );

// To set the preferred orientation with Landscape.
window.SetPreferredOrientation( Landscape );
```                                                                                                                               
> [!NOTE]

> >

Reviewer 2: Global : 

For notes, please remove this one empty line.

> To use `SetPreferredOrientation()`, you must first list up the supported orientation using `AddAvailableOrientation()`. Otherwise, the rotation does not work as expected.
>

## Multi-Window
MultiWIndow means that application use additional window except default window.

<img src="./media/multiwindow3.png">

To Create MultiWindow, the device must support the following features:
- https://www.tizen.org/feature/opengles.surfaceless_context

### Create additional window

It is easy to create an additional window in addition to the default window. The size and position of the new window should be specified. Each window can have its own background color and title:

```csharp
Window newWindow = new Window(new Rectangle(0, 0, 1920, 1080))
{
    BackgroundColor = Color.White,
    Title = "new window"
};
```

> **Note**
>
> To support rotation in multi window, the rotation functions should be used for each window.

> >

Reviewer: Please use markdown tag

and suggest changing to:

To support rotation in multi-window, the [rotation](#rotation) function must be used for each window.

Reference: https://github.com/Samsung/tizen-docs/pull/1170#discussion_r451808564

Reviewer 2: @surbhig12 Would you please check the link in your suggestion also? For now, the \[rotation](#rotation) is going to nowhere.

So the hyperlink is suggested to be removed, or changed to use actual anchor.

## Change window stacking order

NUI provides support for changing the stacking order of the windows.

A window can be raised to the top of the window stack so that no sibling window obscures it:

```csharp
window.Raise();
```

A window can be lowered to the bottom of the window stack so that it does not obscure any sibling windows:

```csharp
window.Lower();
```

New windows are automatically placed to the top of the window stack at creation time by the window manager.

### Set parent window

> To set transient relationship between two windows, use `SetParent()`. After setting the relationship, the two windows are together raised-up, lowered, and iconified or deiconified. Initially, the child window is located on top of the parent. The window can go below parent by calling `Lower()` :

Reviewer: ```suggestion
To set transient relationship between two windows, use `SetParent()`. After setting the relationship, the two windows are together raised-up, lowered, and iconified or deiconified. Initially, the child window is located on top of the parent. The window can go below parent by calling `Lower()`:
```

```csharp
newWindow.Destroy();
```

## Resize window

The size of a window can be retrieved, and the window can be resized.

```csharp
// Retrieve the current size of the window
Size2D windowSize = window.WindowSize;

// Resize the window by increasing its width
windowSize.Width += 100;
window.WindowSize = windowSize;
```

When the window size is changed, a resize event is emitted.

## Move window

It is easy to retrieve the current position of a window and move the window to a new position.

```csharp
// Retrieve the current position of the window
Position2D windowPosition = window.WindowPosition;

// Moving the window to the right
windowPosition.X += 100;
window.WindowPosition = windowPosition;
```

## Show and hide window

When a window is created, by default it is in visible state. It is also possible to hide a window by making it invisible.

```csharp
// Hide the window
window.Hide();

// Show the window
window.Show();
```

If all the windows are hidden, updating and rendering of the windows is paused.

## Event Handling

It allows you to easily handle various events emitted by the window.
The following are some default events:

### Touch event

> Touch event is triggered when the window is touched. if there are multiple touch points, then this will be triggered when the first touch occurs and then when the last finger is lifted.

Reviewer: Yes, it is present tense. Could we suggest:

If you touch the window, the touch event is triggered. If multi points are touched, touch event is triggered when the first touch occurs and then when the last finger is lifted.

> Touch event is triggered when the window is touched. if there are multiple touch points, then this will be triggered when the first touch occurs and then when the last finger is lifted.

Reviewer 2: Thanks for rewriting, @surbhig12.

> Touch event is triggered when the window is touched. if there are multiple touch points, then this will be triggered when the first touch occurs and then when the last finger is lifted.

Reviewer: @tscholb, please incorporate this comment: https://github.com/Samsung/tizen-docs/pull/1170/files#r462030261

    {
        // The window has been touched, do something
    }
}
```

### Key event

Key event is emitted when the window receives a key event from the window manager.

The following code shows how to handle the key event:

```csharp
window.KeyEvent += OnWindowKeyEvent;

private void OnWindowKeyEvent(object sender, Window.KeyEventArgs e)
{
    if (e.Key.State == Key.StateType.Down)
    {
        if (e.Key.KeyPressedName == "Left")
        {
            // The left arrow key is pressed, do something
        }
        else if (e.Key.KeyPressedName == "Right")
        {
            // The right arrow key is pressed, do something
        }
    }
}
```

### Resize event

Resize event is emitted when the window is resized.

The following code shows how to handle the resize event:

```csharp
window.Resized += OnWindowResized;

private void OnWindowResized(object sender, Window.ResizedEventArgs e)
{
    // Window is resized, do something
}
```

For more information, see [Window](https://samsung.github.io/TizenFX/latest/api/Tizen.NUI.BaseComponents.View.html) API.

## Window rotation

To support rotation, NUI provides two methods.
// To do: tscholb

> ### Enable or disable rotation

Reviewer: @tscholb,

As pointed out by @mijin85cho,
Request to please update to Enable or disable window

If the device supports rotation, application can rotate the window with specific orientation.

The supported orientation are `Portriat`, `Landscape`, `PortraitInverse` and `LandscapeInverse`.
`Portrait` means the height is larger than the width or same.
`Landscape` means the width is larger than the height.

<img src="./media/device_rotate2.png">

> Application can enable or disable a specifc orientation to use orientation functions. The following code shows that `Portrait`, `Landscape`, `PortraitInverse` and`LandscapeInverse` are supported in the application.

Reviewer 2: ```suggestion
Application can enable or disable a specific orientation to use orientation functions. The following code shows that `Portrait`, `Landscape`, `PortraitInverse` and `LandscapeInverse` are supported in the application.
```

// To enable Landscape
window.AddAvailableOrientation( Landscape );

// To disable PortraitInverse
window.RemoveAvailableOrientation( PortraitInverse );

// To enable LandscapeInverse
window.AddAvailableOrientation( LandscapeInverse );
```

### Preferred orientation

If you want the window to appear in a specific orientation, the preferred orientation function is supported. The preferred orientation is one of the available orientations. For example, as shown in the following code, the available orientation list has `Portrait`, `Landscape`, and `LandscapeInverse`, and the preferred orientation is set to Landscape. This implies that the window will appear in landscape even when the device is rotated with any other orientation.

```csharp
// To enable available orientations
window.AddAvailableOrientation( Portrait );
window.AddAvailableOrientation( Landscape );
window.AddAvailableOrientation( LandscapeInverse );

// To set the preferred orientation with Landscape.
window.SetPreferredOrientation( Landscape );
```                                                                                                                               
> [!NOTE]
>

> > If the Preferred Orientation function is used without listing the supported orientations (`AddAvailableOrientation`), the rotation will not work as expected.

Reviewer: @mijin85cho,
Please let me know if the following suggestion is okay:

To use `SetPreferredOrientation()`, you must first list up the supported orientation using `AddAvailableOrientation()`. Otherwise, the rotation will not work as expected.

> > If the Preferred Orientation function is used without listing the supported orientations (`AddAvailableOrientation`), the rotation will not work as expected.

Reviewer 2: @surbhig12 
One another reason that I changed the sentence is to avoid the word "will". Is there any other suggestion regarding it?

> > If the Preferred Orientation function is used without listing the supported orientations (`AddAvailableOrientation`), the rotation will not work as expected.

Reviewer: @mijin85cho, in that case usage od does is fine.

Suggestion:
To use `SetPreferredOrientation()`, you must first list up the supported orientation using `AddAvailableOrientation()`. Otherwise, the rotation does not work as expected.

>

## Multi-Window
MultiWIndow means that application use additional window except default window.

<img src="./media/multiwindow3.png">

To Create MultiWindow, the device must support the following features:
- https://www.tizen.org/feature/opengles.surfaceless_context

### Create additional window

> It is easy to create an additional window in addition to the default window. The size and position of the new window should be specified. Each window can have its own background color and title:

Reviewer: @mijin85cho, here the focus is on window. There are multiple windows. And the tense is also uture tense, thats is "can have", we cannot say "can has" as it is incorrect.

We can modify to:
It is easy to create an additional window in addition to the default window. The size and position of the new window should be specified. Each window has its own background color and title:

Please let me know your views.

> It is easy to create an additional window in addition to the default window. The size and position of the new window should be specified. Each window can have its own background color and title:

Reviewer 2: Thanks for pointing out it, @surbhig12. I missed that point.

```csharp
Window newWindow = new Window(new Rectangle(0, 0, 1920, 1080))
{
    BackgroundColor = Color.White,
    Title = "new window"
};
```

> **Note**
>

> > To support rotation in multi window, the [rotation](#rotation) functions should be used for each window.

Reviewer 2: The link is not working.

>

## Change window stacking order

NUI provides support for changing the stacking order of the windows.

A window can be raised to the top of the window stack so that no sibling window obscures it:

```csharp
window.Raise();
```

A window can be lowered to the bottom of the window stack so that it does not obscure any sibling windows:

```csharp
window.Lower();
```

New windows are automatically placed to the top of the window stack at creation time by the window manager.

### Set parent window

> To set transient relationship between two windows, use `SetParent()`. After setting the relationship, the two windows together raise-up, lower, and iconified or deiconified. Initially, the child window is located on top of the parent. The window can go below parent by calling `Lower()`:

Reviewer 2: @surbhig12 Would you please check this sentence again? To me below format seems more proper.

... the two windows are together raised-up, lowered, and iconified or deiconified....

Reviewer: @mijin85cho, I agree.
@tscholb,

Suggest changing to:

To set transient relationship between two windows, use `SetParent()`. After setting the relationship, the two windows are together raised-up, lowered, and iconified or deiconified. Initially, the child window is located on top of the parent. The window can go below parent by calling `Lower()`:

```csharp
// Create a new view
View view = new View();

// Add the view to the window
window.Add(view);

// Remove the view from the window
window.Remove(view);
```

Below is a simple example of how to add content to the window:

```csharp
// Get the window instance
Window window = Window.Instance;
window.BackgroundColor = Color.White;

// Create a TextLabel
TextLabel title = new TextLabel("Hello World");
title.Size2D = new Size2D(window.Size.Width, window.Size.Height);
title.HorizontalAlignment = HorizontalAlignment.Center;
title.VerticalAlignment = VerticalAlignment.Center;

// Add the text to the window
window.Add(title);
```

<img src="./media/new_window3.png">

<a name="deleting"></a>
## Deleting window

A window can be deleted, which will also delete all the views inside the window. However, the window cannot be deleted.

```csharp
newWindow.Destroy();
```

<a name="resizing"></a>
## Resizing window

The size of a window can be retrieved, and the window can be resized.

```csharp
// Retrieve the current size of the window
Size2D windowSize = window.WindowSize;

// Resize the window by increasing its width
windowSize.Width += 100;
window.WindowSize = windowSize;
```

> When the window size is changed, a resize eventwill be emitted.

Reviewer: Suggest rephrasing to:

```suggestion
When the window size is changed, a resize event is emitted.
```

> # Window tutorial

Reviewer 2: Also the cover page to this Window tutorial should be updated to add a description and a link.

This tutorial describes how to create and manage windows in NUI, covering the following subjects:

[Overview](#overview)<br>
[Creating Window](#creating)<br>
[Deleting Window](#deleting)<br>
[Resizing Window](#resizing)<br>
[Moving Window](#moving)<br>
[Showing & Hiding Window](#showinghiding)<br>
[Changing Window Stacking Order](#raisinglowering)<br>
[Event Handling](#eventhandling)<br>
[Window Rotation](#rotation)<br>
[Multi Window](#multiwindow)<br>

> 

Reviewer 2: In the docs.tizen.org, it detects the titles of each section and create the ToC box on the right.

So we don't need to list up each section titles like this.
Please remove line 5 ~ line 14.

<a name="overview"></a>
## Window overview

A window contains the visible content of the application.

When an application is created, it automatically creates the [Creating Window](#defaultwindow) for building the main user interface of the application.

[Additional windows](#additionalwindow) can be created (as needed) to display additional content.

The window delivers [various events](#eventhandling) (e.g. key event and touch event) to the application.

<a name="defaultwindow"></a>
## Creating window

When an application is created, the default window will be automatically created:
 ```csharp
 MyApplication myApp = new MyApplication();
 ```

By default, the default window is created in full screen size.

> It also allows to specify the initial size and position of the default window by passing them to the constructor of the application:

Reviewer: Suggest rephrasing to:

You can specify the initial size and position of the default window using the constructor of the application.

Please verify.

title.VerticalAlignment = VerticalAlignment.Center;

// Add the text to the window
window.Add(title);
```

<img src="./media/new_window3.png">

<a name="deleting"></a>
## Deleting window

A window can be deleted, which will also delete all the views inside the window. However, the window cannot be deleted.

```csharp
newWindow.Destroy();
```

<a name="resizing"></a>
## Resizing window

The size of a window can be retrieved, and the window can be resized.

```csharp
// Retrieve the current size of the window
Size2D windowSize = window.WindowSize;

// Resize the window by increasing its width
windowSize.Width += 100;
window.WindowSize = windowSize;
```

When the window size is changed, a [resize event](#resizing) will be emitted.

<a name="moving"></a>
## Moving window

It is easy to retrieve the current position of a window and move the window to a new position.

```csharp
// Retrieve the current position of the window
Position2D windowPosition = window.WindowPosition;

// Moving the window to the right
windowPosition.X += 100;
window.WindowPosition = windowPosition;
```

<a name="showinghiding"></a>
## Showing and hiding window

> Every window is shown after being created. It is possible to hide a window to make it invisible.

Reviewer: When a window is created, by default it is in visible state. It is also possible to hide a window.

Please verify

Reviewer: @mijin85cho,  thank you. I agree with your suggestion.

```csharp
// Hide the window
window.Hide();

// Show the window
window.Show();
```

> If all the windows are hidden, the updating and rendering of the windows will be paused.

Reviewer: Please remove article "the":

```suggestion
If all the windows are hidden, updating and rendering of the windows will be paused.
```

Reviewer: @mijin85cho, thank you. I agree with your suggestion.

<a name="eventhandling"></a>
## Event Handling

It is easy to handle various events emitted by the window.

> Some useful events are introduced in this guide. Please check the Window API documentation for the full list of events.

Reviewer: @tscholb,

Please incorporate this comment. (https://github.com/Samsung/tizen-docs/pull/1170/files#r454201578)

### Touch event

> Touch event is emitted when the window is touched. If there are multiple touch points, then this will be emitted when the first touch occurs and then when the last finger is lifted.

Reviewer: Suggestion: Can we use triggered instead of emitted?

Touch event is triggered.....

Reviewer 2: Also a question.
According to this description, regardless of how many points are touched, only 2 touch events are emitted?
(when the first touch occurs / when the last touch is lifted)

Below is an example of how to handle the key event:

```csharp
window.KeyEvent += OnWindowKeyEvent;

private void OnWindowKeyEvent(object sender, Window.KeyEventArgs e)
{
    if (e.Key.State == Key.StateType.Down)
    {
        if (e.Key.KeyPressedName == "Left")
        {
            // The left arrow key is pressed, do something
        }
        else if (e.Key.KeyPressedName == "Right")
        {
            // The right arrow key is pressed, do something
        }
    }
}
```

### Resize event

Resize event is emitted when the window is resized.

Below is an example of how to handle the resize event:

```csharp
window.Resized += OnWindowResized;

private void OnWindowResized(object sender, Window.ResizedEventArgs e)
{
    // Window is resized, do something
}
```

<a name="Rotation"></a>
## Window Rotation

To support rotation, NUI provides two methods.

### Enable/Disable rotation

If the device supports rotation, application can rotate the window with specific orientation.

The supported orientation are Portriat, Landscape, PortraitInverse and LandscapeInverse.
Portrait means the width is larger than the height or same.

> Otherwise Landscape means the width is larger than the height.

Reviewer 2: Also would you please check the width and height again?
Comparing to the image, to me, it seems the width and height in the description for portrait are changed.

> <img src="./media/device_rotate2.png">

Reviewer: Please use code tag for Portriat, Landscape, PortraitInverse and LandscapeInverse.

After that, the window will only work Landscape although device is rotated with any orientation.

```csharp
// To enable available orientations
window.AddAvailableOrientation( Portrait );
window.AddAvailableOrientation( Landscape );
window.AddAvailableOrientation( LandscapeInverse );

// To set the preferred orientation with Landscape.
window.SetPreferredOrientation( Landscape );
```                                                                                                                               

> **Note**
>
> If the Preferred Orientation's function is used without the available orientation list, the rotation will not work normal.
>

<a name="multiwindow"></a>
## Multi Window

<a name="additionalwindow"></a>
### Creating Additional Window

It is easy to create an additional window in addition to the default window. The size and position of the new window should be specified. Each window can has its own background color and title.

```csharp
Window newWindow = new Window(new Rectangle(0, 0, 1920, 1080))
{
    BackgroundColor = Color.White,
    Title = "new window"
};
```

> <img src="./media/multiwindow3.png">

Reviewer 2: I can see the image and the sample code doesn't match.
It would be better if you can use the sample code for that image.

A window can be raised to the top of the window stack so that no sibling window obscures it.

```csharp
window.Raise();
```

A window can be lowered to the bottom of the window stack so that it does not obscure any sibling windows.

```csharp
window.Lower();
```

New windows are automatically put to the top of the window stack at creation time by the window manager.

<a name="transientfor"></a>
### Setting parent window

To set transient for two windows, SetParent function is useful.
After setting that, these windows do together when raise-up, lower and iconified/deiconified.
Initially, the window is located on top of the parent. The window can go below parent by calling Lower().
```csharp
// Get the parent window instance
Window parent = Window.Instance;

Window child = new Window(new Rectangle(0, 0, 960, 1080))
{
    BackgroundColor = Color.White,
    Title = "child window"
};

child.SetParent( parent );
```

To unset transient for relationship, call Unparent().
```csharp
child.Unparent();
```

> **Note**
>
> If parent's window stack is changed by calling Raise() or Lower(),> child windows are located on top of the parent again.
>

> 

Reviewer 2: Please add "Related information" section.

## Related Information
- Dependencies
  - Tizen 5.5 and Higher
